File: 202103/ListView/MainActivity.java
package com.example.myapplication;

import android.os.Bundle;
import android.view.View;
import android.widget.ArrayAdapter;
import android.widget.ListView;

import androidx.appcompat.app.AppCompatActivity;

import java.time.LocalTime;
import java.util.ArrayList;

public class MainActivity extends AppCompatActivity {

    private ArrayAdapter<String> mAdapter = null;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        mAdapter = new ArrayAdapter<>(this, android.R.layout.simple_list_item_1, new ArrayList<>());
        ((ListView)findViewById(R.id.listView)).setAdapter(mAdapter);
    }

    /** ボタンが押された */
    public void onClick(View v) {
        //押下時のタイムスタンプを追加していく
        mAdapter.add(LocalTime.now().toString());
    }

}

File: 202103/ListView/MainActivity.kt
package com.example.myapplication


import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.View
import android.widget.ArrayAdapter
import android.widget.ListView
import java.time.LocalTime

class MainActivity : AppCompatActivity() {
    // 引数に渡すListはミュータブル(mutableListOf)であること。
    // listOf()でイミュータブルのリストを渡すと、データ追加できなくなる。
    private val adapter by lazy { ArrayAdapter<String>(this, android.R.layout.simple_list_item_1, mutableListOf()) }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        findViewById<ListView>(R.id.listView).adapter = adapter
    }

    /** ボタンが押された */
    fun onClick(v: View) {
        //押下時のタイムスタンプを追加していく
        adapter.add(LocalTime.now().toString())
    }
}
